client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/iteration/RemotePublisher.java

```diff
@@ -55,15 +56,15 @@ public RemotePublisher(OperationsFactory operationsFactory, String filterConvert
       if (segments == null) {
          if (segmentConsistentHash != null) {
             int maxSegment = segmentConsistentHash.getNumSegments();
-            this.segments = IntSets.mutableEmptySet(maxSegment);
+            this.segments = IntSets.concurrentSet(maxSegment);
             for (int i = 0; i < maxSegment; ++i) {
                this.segments.set(i);
             }
          } else {
             this.segments = null;
          }
       } else {
-         this.segments = IntSets.mutableCopyFrom(segments);
+         this.segments = IntSets.concurrentCopyFrom(IntSets.from(segments), Collections.max(segments) + 1);
```

Comment:
This line is slightly wrong. From what I remember the number of segments provided don't always have to include all segments, so this can cause an issue where the client may only request 5 segments but they are say `0,  6, 28, 76, 125` and it would cause an initialization exception. I would say to be safe we should specify the maxSegment like the other branch.

Reply:
Sorry, I misread, this line is fine. However, I would say it might be simpler to just use maxSegment though.